### PR TITLE
Add Cross DAC documentation

### PR DESCRIPTION
**Pull request by sdmaclea**
Fixes dotnet/diagnostics#1384

### CODE REVIEW

**Review by tommcdon — APPROVED** [review]
Thanks! LGTM

File: docs/design/features/cross-dac.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+The DAC is essentially a memory parsing tool with supporting functionality. The layout of types in the DAC must match the layout of types in the runtime.
+
+The `C++` standard is not explicit about all layout rules of data structures. Due to its historical evolution from `C`, most structures are arranged in an intuitive easily understood fashion. Newer and more exotic structures are less consistent.
+
+Experimentation has shown that layout varies in inheritance cases. The DAC does not support general multiple inheritance, so that simplifies things. It does support multiple inheritance with the empty base classes.
+
+These cases have proven to be problematic:
+
+- Classes with empty base classes. (I the only issue is with multiple base classes.)
```
Comment (k15tfu) [inline docs/design/features/cross-dac.md:43]: typo: `I the only issue is `

File: docs/design/features/cross-dac.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+I use the term `DAC` in this document to refer to both the `DAC` and the `DBI` debug interface. Both were actually cross compiled. Be aware.
+
+### Build entry point
+
+The main build systme change is adding the ability to set the Target OS on a Windows build.
+
+- See [build-runtime.cmd changes](https://github.com/dotnet/runtime/blob/0049c629381c5a18e4dadd1038c2bd6b3ae6e3e6/src/coreclr/build-runtime.cmd#L133-L134)
+- See [Subsets.props](https://github.com/dotnet/runtime/blob/0049c629381c5a18e4dadd1038c2bd6b3ae6e3e6/eng/Subsets.props#L191-L197)
+
+There are also changes to the official build to set these flags package the results and upload to the symbol server.
```
Comment (k15tfu) [inline docs/design/features/cross-dac.md:93]: missing comma: `to set these flags package`

**Review by k15tfu — COMMENTED** [review]
Hi @sdmaclea! Thanks for this doc.

**Pull request merged**